pkg/internal/executors/executor.go

```diff
@@ -49,7 +50,22 @@ type javaExecutorFactory interface {
 	NewJavaExecutor(JavaExecutorOptions) (*JavaExecutor, error)
 }
 
-func NewJavaExecutor(opts JavaExecutorOptions) (*JavaExecutor, error) {
+func NewJavaExecutor(opts JavaExecutorOptions, attachDebugger bool) (*JavaExecutor, error) {
```

Comment:
I think if we can move it into `JavaExecutorOptions`, that'd be nice because we could then add a doc comment that explains that if you set this, we are going to pick Maven for you.

Reply:
I added a comment to this function to specify that.